### PR TITLE
PrintDialog / CompareDialog refactoring

### PR DESCRIPTION
**Pull request by bsongis**

### CODE REVIEW

File: companion/src/modeledit/setup.cpp [hunk 1]
```diff
@@ -35,10 +36,11 @@ TimerPanel::TimerPanel(QWidget *parent, ModelData & model, TimerData & timer, Ge
...
 
   ui->countdownBeep->setField(timer.countdownBeep, this);
-  ui->countdownBeep->addItem(tr("Silent"), 0);
-  ui->countdownBeep->addItem(tr("Beeps"), 1);
+  ui->countdownBeep->addItem(tr("Silent"), TimerData::COUNTDOWN_SILENT);
+  ui->countdownBeep->addItem(tr("Beeps"), TimerData::COUNTDOWN_BEEPS);
   if (IS_ARM(board) || IS_2560(board)) {
-    ui->countdownBeep->addItem(tr("Voice"), 2);
+    ui->countdownBeep->addItem(tr("Voice"), TimerData::COUNTDOWN_VOICE);
+    ui->countdownBeep->addItem(tr("Haptic"), TimerData::COUNTDOWN_HAPTIC);
```
Comment (projectkk2glider) [inline companion/src/modeledit/setup.cpp]: I see we had a missing option here.

**Comment by projectkk2glider** [comment]
I reviewed the code changes, it all looks very good. I've spotted a couple of places that need cosmetic changes, not in your changes but adjacent to them (I already have commits ready).

I will now start compiling and testing the new code.

**Comment by bsongis** [comment]
Thanks Damjan, you may go on in my branch if it's easier for you. The
Haptic option is related to the other commit I did today, in fact when I
started adding it to Companion I realized it was time for some refactoring
and didn't split the commits...
Le 23 août 2015 11:55 AM, "Damjan Adamic" <notifications@github.com> a
écrit :

**Comment by projectkk2glider** [comment]
I've had to do some changes. I hope you don't mind. The first commit are just some cosmetics, basically unrelated to this pull. The second commit contains my fixes and changes.

This is now working OK for me and my models. I will merge after you say it is OK.

**Comment by projectkk2glider** [comment]
One comment: I don't like the new flight modes string:
```
MAX Weight(+100%) Flight modes(FM0 "Normal", FM3 "Thermal", FM4 "Speed") NoTrim [Fixed]
```

the old format is more compact:
```
MAX Weight(+100%) Flight modes(Normal, Thermal, Speed) NoTrim [Fixed]
```

**Comment by bsongis** [comment]
No problem, may I let you change it?

**Comment by projectkk2glider** [comment]
This pull fixes #2737

**Pull request merged**